lib/backend/dynamo/dynamodbbk.go

```diff
@@ -262,14 +276,21 @@ func New(ctx context.Context, params backend.Params) (*Backend, error) {
 	b.streams = streams
 
 	// check if the table exists?
-	ts, err := b.getTableStatus(ctx, b.TableName)
+	ts, tableBillingMode, err := b.getTableStatus(ctx, b.TableName)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	switch ts {
 	case tableStatusOK:
-		break
+		if tableBillingMode == dynamodb.BillingModePayPerRequest {
+			cfg.EnableAutoScaling = false
+			l.Debug("ignoring auto_scaling setting as table is in on-demand mode")
```

Comment:
Nit: Should this be info?
```suggestion
			l.Info("Ignoring auto_scaling setting as table is in on-demand mode.")
```

Reply:
+1 for info, it's a one-off startup message and it can help with troubleshooting without having to restart the auth in debug.